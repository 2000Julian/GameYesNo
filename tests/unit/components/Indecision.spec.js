import { shallowMount } from "@vue/test-utils"
import Indecision from '@/components/Indecicion'

describe('Component Indecision', ()=>{

    let wrapper
    let clgSpy

    beforeEach(()=>{
        wrapper = shallowMount( Indecision )

        clgSpy = jest.spyOn( console, 'log' )//hacemos un mock a esto, 
    }) 


    test('Debe de hacer math con el snapchopt', ()=>{
        expect( wrapper.html()).toMatchSnapshot();
    })

    test('Escribir en el input no debe de disparar nada (console.log)', async() =>{
        const getAnswerSpy = jest.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')
        await input.setValue('Hola mundo')
        expect( clgSpy ).toHaveBeenCalledTimes(1) //verificamos que haya sido llamado el metodo slgSpy
        expect( getAnswerSpy ).not.toHaveBeenCalled()
    })

    test('Al escribir el simbolo de (?) debe de disparar el fetch ', async() =>{
        
        const getAnswerSpy = jest.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')
        await input.setValue('Hola mundo?')

        expect( clgSpy ).toHaveBeenCalledTimes(2) //verificamos que haya sido llamado el metodo slgSpy
        expect( getAnswerSpy ).not.toHaveBeenCalled()
    })

    test('Prueba de getAnswer',() =>{

    })

    test(' Pruebas en getAnswer - Fallo en el API',() =>{

    })
})