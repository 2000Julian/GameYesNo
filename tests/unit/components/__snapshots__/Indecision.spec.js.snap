// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Indecision Debe de hacer math con el snapchopt 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hasme una pregunta">
  <p>Recuerda terminar con un signo de interrogacion (?)</p>
  <!--v-if-->
</div>
`;
